lib/Game.rb

```diff
@@ -0,0 +1,79 @@
+
+class Game
+    include UserInterface
+    def initialize(board, player1, player2)
+        @board =  board
+        @player_1 = player1
+        @player_2 = player2
+        @current_player = @player_1
+    end
+
+    def play
+        loop do 
+            self.show_board(@board)
+            @current_player.ask_move 
+            break if is_game_over?
+            @current_player = @current_player == @player_1 ? @player_2 : @player_1
+        end
+    end
+
+    def is_game_over?
+        is_victory? || is_draw?
+    end
+
+    private
+        def is_victory?
+            if winner? @current_player.piece
+                self.show_board(@board)
+                self.print_winner(@current_player)
+                true
+            else
+                false
+            end
+        end
+
+        def is_draw?
+            if @board.board_full?
+                self.show_board(@board)
+                self.print_draw
+                true
+            else
+                false
+            end
+        end
+
+        def winner?(piece)
```

Comment:
Is there a way we can simplify determining if a player is the winner without having separate checks for horizontal, vertical and diagonal combinations? Here's a suggestion:

You can have an array of arrays the represent the winning combinations as such:

```
WIN_COMBO = [
      [0, 1, 2],
      [3, 4, 5],
      [6, 7, 8],
      [0, 3, 6],
      [1, 4, 7],
      [2, 5, 8],
      [0, 4, 8],
      [6, 4, 2]
]
```
This also assumes you are compiling a particular player's moves into an array. Then you can check this player's compiled moves array agains this win combo array to see if all the player's moves are inside any one of the inner combo arrays in WIN_COMBO. You can use `any?` and `all?` array methods to accomplish this. This will make your winner checking much simpler and cleaner and your game class much smaller.